v2/app/src/main/java/com/google/firebase/example/makeitso/data/repository/AuthRepository.kt

```diff
@@ -0,0 +1,29 @@
+package com.google.firebase.example.makeitso.data.repository
+
+import com.google.firebase.example.makeitso.data.datasource.AuthRemoteDataSource
+import com.google.firebase.example.makeitso.data.model.User
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+class AuthRepository @Inject constructor(
+    private val authRemoteDataSource: AuthRemoteDataSource
+) {
+    val currentUser: Flow<User> = authRemoteDataSource.currentUser
+
+    suspend fun createGuestAccount() {
+        authRemoteDataSource.createGuestAccount()
+    }
+
+    suspend fun signIn(email: String, password: String) {
+        authRemoteDataSource.signIn(email, password)
+    }
+
+    suspend fun signUp(email: String, password: String) {
+       authRemoteDataSource.linkAccount(email, password)
+    }
+
+    suspend fun signOut() {
+        authRemoteDataSource.signOut()
+        authRemoteDataSource.createGuestAccount()
+    }
```

Comment:
This might be misleading since the function name is `signOut` but the function does 2 things: sign out and create a guest account.

Should we maybe perform this logic on the ViewModel rather than the Repository?

Reply:
Yes, great point. I'll do it in the ViewModel.